crates/uv/src/settings.rs

```diff
@@ -73,11 +73,20 @@ impl GlobalSettings {
         Self {
             quiet: args.quiet,
             verbose: args.verbose,
-            color: if args.no_color
-                || std::env::var_os("NO_COLOR")
-                    .filter(|v| !v.is_empty())
-                    .is_some()
+            color: if args.no_color {
+                // If the `--no-color` flag is passed, disable color output.
+                ColorChoice::Never
+            } else if matches!(
+                args.color,
+                ColorChoice::Always | ColorChoice::Never | ColorChoice::Auto
+            ) {
+                // If `--color` is passed explicitly, use the value of `args.color` directly.
+                args.color
```

Comment:
Won't this always match since the default is "auto" at the clap-level?

Reply:
Yupp, But I thought if `--color` is passed explicitly, we can use the value of `args.color` directly.

Reply:
Also, when I tested the expected result was coming for all three separate cases. Means for `FORCE_COLOR` and `--color` and with sub arguments.

Reply:
> Won't this always match since the default is "auto" at the clap-level?

Done purposely, as when the user does not anything (neither the environment variables nor the --color), then I wanted the auto property. As If `--color` is given, it doesn't matter the value, we can always use them.